packages/turf-dissolve/index.ts

```diff
@@ -25,57 +26,70 @@ import polygonClipping from "polygon-clipping";
  * //addToMap
  * var addToMap = [features, dissolved]
  */
-function dissolve(fc, options) {
+function dissolve(
+  fc: FeatureCollection<Polygon>,
+  options: {
+    propertyName?: string;
+  } = {}
+): FeatureCollection<Polygon> {
   // Optional parameters
-  options = options || {};
-  if (!isObject(options)) throw new Error("options is invalid");
-  var propertyName = options.propertyName;
+  const { propertyName } = options;
 
   // Input validation
   collectionOf(fc, "Polygon", "dissolve");
 
   // Main
-  var outFeatures = [];
-  if (!options.propertyName) {
+  const outFeatures = [];
+  if (!propertyName) {
     return flatten(
       multiPolygon(
         polygonClipping.union.apply(
           null,
+          // List of polygons expressed as Position[][][] a.k.a. Geom[]
           fc.features.map(function (f) {
             return f.geometry.coordinates;
-          })
+          }) as [Geom]
```

Comment:
Have updated it to ```[Geom, ...Geom[]]```. The interface to union() expects an initial Geom, then any extra Geoms are spread in as rest parameters (rather than an array).